libpod/image.go

```diff
@@ -50,29 +55,116 @@ type CopyOptions struct {
 // return will include the image, a false return will exclude it.
 type ImageFilter func(*storage.Image) bool
 
+type imagePullStruct struct {
+	imageName   string
+	tag         string
+	registry    string
+	hasRegistry bool
+	transport   string
+}
+
+func (ips imagePullStruct) returnFQName() string {
+	return fmt.Sprintf("%s%s/%s:%s", ips.transport, ips.registry, ips.imageName, ips.tag)
+
+}
+
+func getRegistriesToTry(image string) ([]types.ImageReference, error) {
+	var registries []types.ImageReference
+	var imageError = fmt.Sprintf("Unable to parse '%s'\n", image)
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return registries, errors.Errorf(imageError)
+	}
+
+	tagged, istagged := imgRef.(reference.NamedTagged)
+	tag := "latest"
+	if istagged {
+		tag = tagged.Tag()
+	}
+	hasDomain := true
+	registry := reference.Domain(imgRef.(reference.Named))
+	if registry == "" {
+		hasDomain = false
+	}
+	imageName := reference.Path(imgRef.(reference.Named))
+	pImage := imagePullStruct{
+		imageName,
+		tag,
+		registry,
+		hasDomain,
+		DefaultRegistry,
+	}
+	if pImage.hasRegistry {
+		// If input has a registry, we have to assume they included an image
+		// name but maybe not a tag
+		pullRef, err := alltransports.ParseImageName(pImage.returnFQName())
+		if err != nil {
+			return registries, errors.Errorf(imageError)
+		}
+		registries = append(registries, pullRef)
+	} else {
+		// No registry means we check the globals registries configuration file
+		// and assemble a list of candidate sources to try
+		registryConfigPath := ""
+		envOverride := os.Getenv("REGISTRIES_CONFIG_PATH")
+		if len(envOverride) > 0 {
+			registryConfigPath = envOverride
+		}
+		searchRegistries, err := sysregistries.GetRegistries(&types.SystemContext{SystemRegistriesConfPath: registryConfigPath})
+		if err != nil {
+			fmt.Println(err)
+			return registries, errors.Errorf("Unable to parse the registries.conf file and"+
+				" the image name '%s' is incomplete.", imageName)
+		}
+		for _, searchRegistry := range searchRegistries {
+			pImage.registry = searchRegistry
+			pullRef, err := alltransports.ParseImageName(pImage.returnFQName())
+			if err != nil {
+				return registries, errors.Errorf(imageError)
+			}
+			registries = append(registries, pullRef)
+		}
+	}
+	return registries, nil
+}
+
+// PullImage copies the image from the source to the destination
+// configured
+func (r *Runtime) PullImage(imgName string, allTags bool, reportWriter io.Writer) error {
+	//func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *types.SystemContext) error {
+	// An eventual array of imagereferences to pull on
+	var fqregistries []types.ImageReference
+	srcRef, err := alltransports.ParseImageName(imgName)
+	if err != nil {
+		fqregistries, err = getRegistriesToTry(imgName)
+		if err != nil {
+			fmt.Println(err)
+		}
+	} else {
+		fqregistries = append(fqregistries, srcRef)
```

Comment:
I think fqregistries will always be empty when this runs, so can't you just do `fqregistries = srcRef` instead?